Hi @awson ,
thanks again for this! In fact this does create a linker problem for the old MinGW build, which is based on Qt-provided MinGW 4.8.2. Not sure if your changes could be introduced without this side effect?

```
 ../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj):hid.c:(.text+0x3a8): undefined reference to `HidD_GetPreparsedData'
../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj):hid.c:(.text+0x456): undefined reference to `HidD_GetSerialNumberString'
../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj):hid.c:(.text+0x578): undefined reference to `HidD_FreePreparsedData'
../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj):hid.c:(.text+0x767): undefined reference to `HidD_GetPreparsedData'
../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj):hid.c:(.text+0x7a6): undefined reference to `HidD_FreePreparsedData'
../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj):hid.c:(.text+0x848): undefined reference to `HidD_FreePreparsedData'
../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj):hid.c:(.text+0x1590): undefined reference to `HidD_GetSerialNumberString'
../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj):hid.c:(.text+0x1657): undefined reference to `HidD_GetIndexedString'
C:/Qt/Tools/mingw482_32/bin/../lib/gcc/i686-w64-mingw32/4.8.2/../../../../i686-w64-mingw32/bin/ld.exe: ../external_libraries/hidapi/windows/libhidapi.a(hid.c.obj): bad reloc address 0x20 in section `.eh_frame'
collect2.exe: error: ld returned 1 exit status
lang\CMakeFiles\sclang.dir\build.make:140: recipe for target 'lang/Release/sclang.exe' failed
mingw32-make.exe[2]: *** [lang/Release/sclang.exe] Error 1
CMakeFiles\Makefile2:2739: recipe for target 'lang/CMakeFiles/sclang.dir/all' failed
mingw32-make.exe[1]: *** [lang/CMakeFiles/sclang.dir/all] Error 2
Makefile:126: recipe for target 'all' failed
mingw32-make.exe: *** [all] Error 2
```

--- a/hidapi_parser/hidapi_parser.c
+++ b/hidapi_parser/hidapi_parser.c
@@ -29,6 +29,9 @@
 
 #ifdef _WIN32
 #include <windows.h>
+// for MinGW < 5.3 include locally provided "../windows hidsdi.h" rather
+// than #include <hidsdi.h>:
+// #include "../windows/hidsdi.h"
 #include <hidsdi.h>
 #endif
 

--- a/windows/hid.c
+++ b/windows/hid.c
@@ -21,6 +21,9 @@
 ********************************************************/
 
 #include <windows.h>
+// for MinGW < 5.3 include locally provided "../windows hidsdi.h" rather
+// than #include <hidsdi.h>:
+// #include "./hidsdi.h"
 #include <hidsdi.h>
 
 #ifndef _NTDEF_

